Let the compiler generate more comparison operators in WebKit/

cdumez opened the pull request:
#### 99ae595c6281e1d6fc153999c8121aa63c3c5039
<pre>
Let the compiler generate more comparison operators in WebKit/
<a href="https://bugs.webkit.org/show_bug.cgi?id=261252">https://bugs.webkit.org/show_bug.cgi?id=261252</a>

Reviewed by Dan Glastonbury.

Let the compiler generate more comparison operators in WebKit/ now that
we support C++20.

* Source/WebGPU/WGSL/SourceSpan.h:
(WGSL::SourceSpan::operator== const): Deleted.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::OperatingDate::operator== const): Deleted.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:
* Source/WebKit/Shared/LayerTreeContext.cpp:
(WebKit::LayerTreeContext::LayerTreeContext): Deleted.
(WebKit::LayerTreeContext::~LayerTreeContext): Deleted.
(WebKit::operator==): Deleted.
* Source/WebKit/Shared/LayerTreeContext.h:
(WebKit::LayerTreeContext::LayerTreeContext):
* Source/WebKit/Shared/MonotonicObjectIdentifier.h:
(WebKit::MonotonicObjectIdentifier::operator== const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
(WebKit::RemoteLayerBackingStore::Parameters::operator== const): Deleted.
* Source/WebKit/Shared/ScrollingAccelerationCurve.h:
(WebKit::ScrollingAccelerationCurve::operator==):
(WebKit::ScrollingAccelerationCurve::operator== const): Deleted.
* Source/WebKit/Shared/WebPageNetworkParameters.h:
(WebKit::WebPageNetworkParameters::operator== const): Deleted.
* Source/WebKit/Shared/glib/InputMethodState.h:
(WebKit::operator==): Deleted.

Canonical link: <a href="https://commits.webkit.org/267751@main">https://commits.webkit.org/267751@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b03e8300b25834d8987723d40dd12258c3e0f416

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17599 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17924 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18453 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19407 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16457 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21209 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18070 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17809 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18111 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15278 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/20262 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15346 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22628 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16355 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16192 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20484 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16764 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/14198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15882 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4190 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20248 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16606 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebKit/Shared/LayerTreeContext.cpp:
```diff
@@ -32,23 +32,13 @@
-LayerTreeContext::~LayerTreeContext()
-{
-}
+LayerTreeContext::~LayerTreeContext() = default;
```
Do we even need to spell this out? Why not just delete the code?

cdumez replied:
I wasn’t sure what the intention was. If I remove the code then the constructor and destructor will be inline. It seems like whoever wrote this went out of their way to move them out of line so I kept that.

That said, I don’t really see the point of not having them inline so I’ll drop the implementation unless someone tells me otherwise.

darinadler replied:
Typically we move them out of line when we can compile the header with fewer includes that way.

cdumez replied:
Yes, though definitely not the case here.

djg commented on Source/WebKit/Shared/LayerTreeContext.h:
```diff
@@ -40,16 +40,16 @@ enum class LayerHostingMode : uint8_t {
     LayerTreeContext();
-    LayerTreeContext(uint64_t id) : contextID(id) { };
+    LayerTreeContext(uint64_t id) : contextID(id) { }
     ~LayerTreeContext();
```
Ditto

djg reviewed (approved):
:shipit:

cdumez committed:
Let the compiler generate more comparison operators in WebKit/
https://bugs.webkit.org/show_bug.cgi?id=261252

Reviewed by Dan Glastonbury.

Let the compiler generate more comparison operators in WebKit/ now that
we support C++20.

* Source/WebGPU/WGSL/SourceSpan.h:
(WGSL::SourceSpan::operator== const): Deleted.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::OperatingDate::operator== const): Deleted.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:
* Source/WebKit/Shared/LayerTreeContext.cpp:
(WebKit::LayerTreeContext::LayerTreeContext): Deleted.
(WebKit::LayerTreeContext::~LayerTreeContext): Deleted.
(WebKit::operator==): Deleted.
* Source/WebKit/Shared/LayerTreeContext.h:
(WebKit::LayerTreeContext::LayerTreeContext):
* Source/WebKit/Shared/MonotonicObjectIdentifier.h:
(WebKit::MonotonicObjectIdentifier::operator== const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
(WebKit::RemoteLayerBackingStore::Parameters::operator== const): Deleted.
* Source/WebKit/Shared/ScrollingAccelerationCurve.h:
(WebKit::ScrollingAccelerationCurve::operator==):
(WebKit::ScrollingAccelerationCurve::operator== const): Deleted.
* Source/WebKit/Shared/WebPageNetworkParameters.h:
(WebKit::WebPageNetworkParameters::operator== const): Deleted.
* Source/WebKit/Shared/glib/InputMethodState.h:
(WebKit::operator==): Deleted.

Canonical link: https://commits.webkit.org/267751@main
